server/controllers/card.js

```diff
@@ -1,98 +1,117 @@
-  const {columnId, boardId, title, tag} = req.body
-  const column = await Column.findOne({_id: columnId, boardId: boardId})
+exports.createCard = asyncHandler(async (req, res, next) => {
+  const { columnId, boardId, title, tag, deadline, description } = req.body;
+  const column = await Column.findOne({ _id: columnId, boardId: boardId });
   if (!column) {
     res.status(404);
     throw new Error("No Column found");
   }
-  const files = req.files
-  let images = []
-  if(files.length) {
-    images = await Promise.all (files.map(async file => {
-      const result = await uploadFile(file)
-      await unlinkFile(file.path)
-      return result.key
-    }))
+  const files = req.files;
+  let images = [];
+  if (files.length) {
+    images = await Promise.all(
+      files.map(async (file) => {
+        const result = await uploadFile(file);
+        await unlinkFile(file.path);
+        return result.key;
+      })
+    );
   }
   const newCard = await Card.create({
     title,
     tag,
+    description: description,
+    deadline: deadline,
+    description,
+    deadline,
     columnId: column._id,
-    images : [...images]
-  })
-  column.cards.push(newCard)
-  column.save()
-  res.status(200).json(newCard)
-})
-
-exports.updateCard = asyncHandler(async (req, res)=> {
-  const card = await Card.findById(req.params.id)
+    boardId: boardId,
+    images: [...images],
+  });
+  column.cards.push(newCard);
+  column.save();
+  res.status(200).json(newCard);
+});
+exports.updateCard = asyncHandler(async (req, res) => {
+  const card = await Card.findById(req.params.id);
   if (!card) {
     res.status(404);
     throw new Error("No Card found");
   }
 
-  const {currentImages, title, tag} = req.body
+  const { currentImages, title, tag, deadline, description } = req.body;
   // currentImages is an array containing the keys of current images in the card
   // update the images in db based on the current images in the card
-  if(card.images.length && currentImages.length) {
-    await deleteImagesInS3(card.images, currentImages)
+  if (card.images.length && currentImages.length) {
+    await deleteImagesInS3(card.images, currentImages);
   }
-  
-  const files = req.files
-  let images = []
+  const files = req.files;
+  let images = [];
   // if user additionally upload new image, save them as well
-  if(files.length) {
-    images = await Promise.all (files.map(async file => {
-      const result = await uploadFile(file)
-      await unlinkFile(file.path)
-      return result.key
-    }))
+  if (files.length) {
+    images = await Promise.all(
+      files.map(async (file) => {
+        const result = await uploadFile(file);
+        await unlinkFile(file.path);
+        return result.key;
+      })
+    );
   }
 
-  card.title = title
-  card.tag  = tag
-  card.images = [...currentImages, ...images]
-  card.save()
-  res.status(200).json(card)
-})
-
+  if (tag) {
+    card.tag = tag;
```

Comment:
You can also use a one liner for this if (1) s=5 or a loop

Reply:
Thanks for the suggestion. updated the code.